test/alternator/test_streams.py

```diff
@@ -1581,6 +1581,26 @@ def test_stream_arn_unchanging(dynamodb, dynamodbstreams):
         assert len(streams['Streams']) == 1
         assert streams['Streams'][0]['StreamArn'] == arn
 
+# Enabling a stream shouldn't cause any extra table to appear in ListTables.
+# In issue #19911, enabling streams on a table called xyz caused the name
+# "xyz_scylla_cdc_log" to appear in ListTables. The following test creates
+# a table with a long unique name, and ensures that only one table containing
+# this name as a substring is listed.
+# In test_gsi.py and test_lsi.py we have similar tests for GSI and LSI.
+# Reproduces #19911
+def test_stream_list_tables(dynamodb):
+    with new_test_table(dynamodb,
+        Tags=TAGS,
+        StreamSpecification={'StreamEnabled': True, 'StreamViewType': 'KEYS_ONLY'},
+        KeySchema=[ { 'AttributeName': 'p', 'KeyType': 'HASH' } ],
+        AttributeDefinitions=[ { 'AttributeName': 'p', 'AttributeType': 'S' }, ]
+    ) as table:
+            # Check that the long and unique table name (created by
+            # unique_table_name()) isn't a substring of any table name,
+            # except of course the table itself:
+            for listed_name in list_tables(dynamodb):
+                assert table.name == listed_name or table.name not in listed_name
```

Comment:
If `table.name == listed_name` is always false and `table.name not in listed_name` always true, the test passes. I am not sure how this could happen, only if `list_tables` stops working? Anyways, this check opens a possibility for this testcase to not test what it is supposed to in the future, I think.
Maybe we can split these checks into something like:
```
assert table.name in list_tables(dynamodb)
for listed_name in list_tables(dynamodb):
    assert table.name not in listed_name

Reply:
This test reproduces the bug that enabling streams suddenly adds tablename_scylla_cdc_log_table to ListTables. You're saying theoretically we could have a _different_ bug where adding a stream the original table suddenly disappears from ListTables (because before adding a stream, we do already have a test that ListTable works). Unlikely (and not the bug I was trying to reproduce), but I guess wouldn't hurt to check that too. I'll send a new version.

Reply:
Oh, I see that @xemul already merged. I'll send a followup.

Reply:
> ```
> assert table.name in list_tables(dynamodb)
> for listed_name in list_tables(dynamodb):
>     assert table.name not in listed_name
> ```

By the way, the second assert is wrong because the table **itself** is one of these listed_name. I do need to allow the table.name == listed_name case too.

Reply:
Oh, right! Is there an option to remove an element from a python's list so that if this element is missing, then the code throws? If such function exists, my suggestion could be rewritten as:
```
tables = list_tables(dynamodb)
tables.remove(table.name)
for listed_name in tables:
    assert table.name not in listed_name

Reply:
https://github.com/scylladb/scylladb/pull/19934

Reply:
@ptrsmrn your new suggestion again forgets to assert before the remove that table.name is in tables, so it's another line.
Please take a look at what I did in #19994. I used an "if". I know it's old-school, but I think it's actually the clearest code.

Reply:
> your new suggestion again forgets to assert before the remove that table.name is in tables, so it's another line.

@nyh as I wrote in my comment above, I assumed that such a python function would throw in case a to-be-removed element doesn't exist. But if such a throwing function doesn't exist, then yes, we'd need to assert before removing.

Reply:
I see that `list.remove(x)` is already working that way and if the element doesn’t exist, it throws ValueError: list.remove(x): x not in list exception, so IMO the assertion is not needed (because we get an exception thrown). But that's just FYI, no need to change any code.